src/main/java/com/cse/helium/app/utils/InvalidParameterResponses.java

```diff
@@ -0,0 +1,125 @@
+      VALIDATION_ERROR_TEMPLATE,
+      "PageSize",
+      "The parameter 'PageSize' should be between 1 and 1000.");
+
+  private static final String PAGENUM_ERROR = String.format(
+      VALIDATION_ERROR_TEMPLATE,
+      "PageNumber",
+      "The parameter 'PageNumber' should be between 1 and 10000.");
+
+  private static final String GENRE_ERROR = String.format(
+      VALIDATION_ERROR_TEMPLATE,
+      "Genre",
+      "The parameter 'Genre' should be between 3 and 20 characters.");
+
+  private static final String YEAR_ERROR = String.format(
+      VALIDATION_ERROR_TEMPLATE,
+      "Year",
+      "The parameter 'Year' should be between 1874 and 2025.");
+
+  private static final String RATING_ERROR = String.format(
+      VALIDATION_ERROR_TEMPLATE,
+      "Rating",
+      "The parameter 'Rating' should be between 0 and 10.");
+
+  private static final String ACTORID_ERROR = String.format(
+      VALIDATION_ERROR_TEMPLATE,
+      "ActorId",
+      "The parameter 'ActorId' should start with 'nm' and be between 7 and 11 characters in total");
+  
+  private static final String MOVIEID_ERROR = String.format(
+      VALIDATION_ERROR_TEMPLATE,
+      "MovieId",
+      "The parameter 'MovieId' should start with 'tt' and be between 7 and 11 characters in total");
+
+  public enum SearchParameter {
+    Q,
+    PAGE_SIZE,
+    PAGE_NUMBER,
+    YEAR,
+    GENRE,
+    RATING,
+    ACTOR_ID
+  }
+
+  InvalidParameterResponses() {
+  }
+
+  private String parameterToValidationError(SearchParameter parameter) {
+    switch (parameter) {
+      case Q:
+        return Q_ERROR;
+      case PAGE_SIZE:
+        return PAGESIZE_ERROR;
+      case PAGE_NUMBER:
+        return PAGENUM_ERROR;
+      case YEAR:
+        return YEAR_ERROR;
+      case GENRE:
+        return GENRE_ERROR;
+      case RATING:
+        return RATING_ERROR;
+      case ACTOR_ID:
+        return ACTORID_ERROR;
+      default:
+        return "";
+    }
+  }
+
+  private String parametersToValidationErrors(List<SearchParameter> paramters) {
+    List<String> validationErrors = paramters.stream().map(this::parameterToValidationError)
+        .collect(Collectors.toList());
+    return String.join(",\n", validationErrors);
+  }
+
+  private String response(String type, String instance, String validationErrorsBody) {
+    return "{\n"
+      + "  \"type\": \"" + type + "\",\n"
+      + "  \"title\": \"Parameter validation error\",\n"
+      + "  \"detail\": \"One or more invalid parameters were specified.\",\n"
+      + "  \"status\": 400,\n"
+      + "  \"instance\": \"" + instance + "\",\n"
+      + "  \"validationErrors\": [\n"
+      +    validationErrorsBody + "\n"
+      + "  ]\n"
+      + "}";
+  }
+
+  public String invalidActorSearchResponse(String instance, List<SearchParameter> parameters) {
+    return response("https://github.com/retaildevcrews/helium/blob/main/docs/ParameterValidation.md#actors", instance, parametersToValidationErrors(parameters));
```

Comment:
Might be good to isolate this github url into a constant since it is being shared in multiple methods, and then append the tag after.

Reply:
good catch! done